[WIP] AZINTS-2487 Metric filtering updates to terraform (#2479)
DataDog/terraform-provider-datadog: docs/resources/integration_azure.md

Note: This requires `resource_collection_enabled` to be set to true. Defaults to `false`.
- `custom_metrics_enabled` (Boolean) Enable custom metrics for your organization. Defaults to `false`.
- `host_filters` (String) String of host tag(s) (in the form `key:value,key:value`) defines a filter that Datadog will use when collecting metrics from Azure. Limit the Azure instances that are pulled into Datadog by using tags. Only hosts that match one of the defined tags are imported into Datadog. e.x. `env:production,deploymentgroup:red` Defaults to `""`.

> - `metrics_config` (String) A JSON object representing the metrics configuration. It must include the excluded_resource_providers attribute which is a list of Microsoft Resource Provider name strings for which metrics collection is to be disabled. If excluded_resource_providers is an empty list, it means that all Microsoft Resource Providers are enabled for metrics filtering. Defaults to `"{"excluded_resource_providers":[]}"`.

Reviewer: ```suggestion
- `metrics_config` (String) A JSON object representing the metrics configuration. It must include the `excluded_resource_providers` attribute, a list of Microsoft Resource Provider name strings for which metrics collection will be disabled. If `excluded_resource_providers` is an empty list, all Microsoft Resource Providers are enabled for metrics filtering. Defaults to `"{"excluded_resource_providers":[]}"`.
```
